FBTweak/FBTweak.m

```diff
@@ -41,72 +41,83 @@ - (void)encodeWithCoder:(NSCoder *)coder
-    _currentValue = [coder decodeObjectForKey:@"currentValue"] ?: _currentValue;
+@synthesize name = _name;
+@synthesize identifier = _identifier;
+@synthesize currentValue = _currentValue;
+
+
+- (instancetype)initWithIdentifier:(NSString *)identifier name:(NSString *)name
+                      currentValue:(FBTweakValue)currentValue {
+  if (self = [super init]) {
+    _identifier = identifier;
+    _name = name;
+    _currentValue = currentValue;
   }
-  
   return self;
 }
 
-- (instancetype)initWithIdentifier:(NSString *)identifier
+- (void)reset
 {
-  if ((self = [super init])) {
+}
+
+@end
+
+@implementation FBActionTweak
+
+@synthesize name = _name;
+@synthesize identifier = _identifier;
+@synthesize currentValue = _currentValue;
+
+- (instancetype)initWithIdentifier:(NSString *)identifier name:(NSString *)name
+                             block:(dispatch_block_t)block {
+  if (self = [super init]) {
     _identifier = identifier;
-    NSData *archivedValue = [[NSUserDefaults standardUserDefaults] objectForKey:_identifier];
-    _currentValue = (archivedValue != nil && [archivedValue isKindOfClass:[NSData class]] ? [NSKeyedUnarchiver unarchiveObjectWithData:archivedValue] : archivedValue);
+    _name = name;
+    _currentValue = block;
   }
-  
   return self;
 }
 
-- (void)encodeWithCoder:(NSCoder *)coder
+- (void)reset
 {
-  [coder encodeObject:_identifier forKey:@"identifier"];
-  [coder encodeObject:_name forKey:@"name"];
-  
-  if (!self.isAction) {
-    [coder encodeObject:_defaultValue forKey:@"defaultValue"];
-    [coder encodeObject:_possibleValues forKey:@"possibleValues"];
-    [coder encodeObject:_currentValue forKey:@"currentValue"];
-    [coder encodeObject:_precisionValue forKey:@"precisionValue"];
-    [coder encodeObject:_stepValue forKey:@"stepValue"];
-  }
 }
 
-- (BOOL)isAction
-{
-  // NSBlock isn't a public class, walk the hierarchy for it.
-  Class blockClass = [^{} class];
+@end
 
-  while ([blockClass superclass] != [NSObject class]) {
-    blockClass = [blockClass superclass];
+@implementation FBMutableTweak
+
+@synthesize name = _name;
+@synthesize identifier = _identifier;
+@synthesize currentValue = _currentValue;
+@synthesize defaultValue = _defaultValue;
+@synthesize possibleValues = _possibleValues;
+@synthesize precisionValue = _precisionValue;
+@synthesize stepValue = _stepValue;
+
+- (instancetype)initWithIdentifier:(NSString *)identifier name:(NSString *)name
+                      defaultValue:(FBTweakValue)defaultValue {
+  if (self = [super init]) {
+    _identifier = identifier;
+    _name = name;
+    _defaultValue = defaultValue;
+    NSData *archivedValue = [[NSUserDefaults standardUserDefaults] objectForKey:identifier];
+    self.currentValue = (archivedValue != nil && [archivedValue isKindOfClass:[NSData class]] ?
+                         [NSKeyedUnarchiver unarchiveObjectWithData:archivedValue] : archivedValue);
```

Comment:
All the section of `NSUserDefaults` belongs to `FBPersistentTweak` right?

Reply:
Yes, fixed.